cdap-proto/src/main/java/co/cask/cdap/proto/id/NamespacedId.java

```diff
@@ -15,9 +15,16 @@
  */
 package co.cask.cdap.proto.id;
 
+import co.cask.cdap.proto.element.EntityType;
+
 /**
  * An {@link EntityId} which belongs to a namespace.
  */
-public interface NamespacedId {
-  String getNamespace();
+public abstract class NamespacedId extends EntityId {
+
+  protected NamespacedId(EntityType entity) {
+    super(entity);
+  }
+
+  public abstract String getNamespace();
```

Comment:
Why this is abstract method? I think this should be a `final` method, with the constructor always take a `String namespace`. With that, you can also remove all sub-classes `getNamespace` implementations and move the `namespace` field to this class (it shouldn't break the Gson structure by doing so).